src/styles/utils/transitionStyle.tsx

```diff
@@ -0,0 +1,20 @@
+import { css } from "styled-components";
+
+import { ResponsiveValues, responsiveStyle } from "./responsiveStyle";
+
+type PropertyName = "all";
```

Comment:
This interface doesn't make sense to me and I don't like how the types are defined here and not in the theme folder. Let's go for named transitions. Just make one up and make a transitions file in the theme folder.